tools/lambda-promtail/lambda-promtail/main.go

```diff
@@ -54,20 +56,45 @@ func init() {
 	fmt.Println("keep stream: ", keepStream)
 }
 
-func handler(ctx context.Context, ev map[string]interface{}) error {
-	s3event, err := convertToS3Event(ev)
-	if err == nil {
-		return processS3Event(ctx, s3event)
+func checkEventType(ev map[string]interface{}) (interface{}, error) {
+	var s3Event events.S3Event
+	var cwEvent events.CloudwatchLogsEvent
+
+	types := [...]interface{}{&s3Event, &cwEvent}
+
+	j, _ := json.Marshal(ev)
+	reader := strings.NewReader(string(j))
+	d := json.NewDecoder(reader)
+	d.DisallowUnknownFields()
+
+	for _, t := range types {
+		err := d.Decode(t)
+
+		if err == nil {
+			return t, err
```

Comment:
W.e should probably explicitly `return t, nil` here. 

Alternatively you could do:
```
if err != nil {
    reader.Seek(0, 0)
    continue
} 

return t, nil
```